src/lib/root_prover/root_prover.ml

```diff
@@ -0,0 +1,160 @@
+open Protocols
+open Core
+open Async
+open Protocols.Coda_transition_frontier
+open Coda_base
+
+module type Inputs_intf = sig
+  include Transition_frontier.Inputs_intf
+
+  module Transition_frontier :
+    Transition_frontier_intf
+    with type state_hash := State_hash.t
+     and type external_transition_verified := External_transition.Verified.t
+     and type ledger_database := Ledger.Db.t
+     and type staged_ledger := Staged_ledger.t
+     and type masked_ledger := Ledger.Mask.Attached.t
+     and type transaction_snark_scan_state := Staged_ledger.Scan_state.t
+     and type staged_ledger_diff := Staged_ledger_diff.t
+     and type consensus_local_state := Consensus.Local_state.t
+
+  module Time : Protocols.Coda_pow.Time_intf
+
+  module Protocol_state_validator :
+    Protocol_state_validator_intf
+    with type time := Time.t
+     and type state_hash := State_hash.t
+     and type external_transition := External_transition.t
+     and type external_transition_proof_verified :=
+                External_transition.Proof_verified.t
+     and type external_transition_verified := External_transition.Verified.t
+end
+
+module Make (Inputs : Inputs_intf) :
+  Coda_transition_frontier.Root_prover_intf
+  with type state_body_hash := State_body_hash.t
+   and type transition_frontier := Inputs.Transition_frontier.t
+   and type external_transition := Inputs.External_transition.t
+   and type proof_verified_external_transition :=
+              Inputs.External_transition.Proof_verified.t
+   and type consensus_state := Consensus.Consensus_state.value = struct
+  open Inputs
+
+  type t = {logger: Logger.t; finality_length: int}
```

Comment:
Can we make finality length here a functor argument since it is always known at compile time? Given the recent errors we spent a week debugging with compile time vs run time values, I would like to avoid ever passing around values representing compile time parameters so that we can avoid further divergence.

Reply:
If you end up doing that, there is really no point in even having a `t` type since it's just a logger, which can be passed into function calls as an argument. No reason to introduce types when you don't need one.

Reply:
I will make this change and make `max_length` a functor parameter